app/src/main/java/org/greenstand/android/TreeTracker/activities/MainActivity.kt

```diff
@@ -18,25 +18,35 @@ import androidx.core.content.ContextCompat
 import androidx.fragment.app.Fragment
 import androidx.navigation.NavController
 import androidx.navigation.findNavController
-import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.activity_main.toolbar
+import kotlinx.android.synthetic.main.activity_main.toolbarTitle
 import org.greenstand.android.TreeTracker.R
 import org.greenstand.android.TreeTracker.analytics.Analytics
 import org.greenstand.android.TreeTracker.application.Permissions
 import org.greenstand.android.TreeTracker.fragments.DataFragment
 import org.greenstand.android.TreeTracker.fragments.MapsFragmentDirections
 import org.greenstand.android.TreeTracker.managers.FeatureFlags
 import org.greenstand.android.TreeTracker.managers.LanguageSwitcher
-import org.greenstand.android.TreeTracker.managers.UserLocationManager
+import org.greenstand.android.TreeTracker.managers.LocationDataCapturer
+import org.greenstand.android.TreeTracker.managers.LocationUpdateManager
 import org.greenstand.android.TreeTracker.managers.UserManager
 import org.greenstand.android.TreeTracker.utilities.ValueHelper
+import org.koin.android.ext.android.get
 import org.koin.android.ext.android.inject
 
 class MainActivity : AppCompatActivity(), ActivityCompat.OnRequestPermissionsResultCallback {
 
     private val languageSwitcher: LanguageSwitcher by inject()
     private val userManager: UserManager by inject()
     private val analytics: Analytics by inject()
-    private val userLocationManager: UserLocationManager by inject()
+    private val locationUpdateManager: LocationUpdateManager by inject()
+
+    // An eager instance of LocationDataCapturer assures the location data capture for the duration
+    // the Treetracker App is in the foreground as required for data analysis. Even if this instance
+    // is not used explicitly in this activity, this declaration is necessary to initiate the
+    // process.
+    private val locationDataCapturer: LocationDataCapturer = get()
```

Comment:
Why not just inject lazily and call the start method?

Reply:
I considered it as well. Will change this to have a start method, I am ok with that.
Just curious, is there any reason to not use KoinComponent other than the preference for a simple pojo?

Reply:
Updated the PR with this change, this choice is more explicit. Thank you for the feedback @Elforama